### PR TITLE
Implement UpdateWorker service method

### PR DESCRIPTION
**Pull request by talanknight**

### CODE REVIEW

File: internal/daemon/controller/handlers/workers/worker_service_test.go [hunk 1]
```diff
@@ -284,3 +287,411 @@ func TestDelete(t *testing.T) {
...
+				require.Error(t, gErr)
+				assert.True(t, errors.Is(gErr, tc.err), "UpdateWorker(%+v) got error %v, wanted %v", req, gErr, tc.err)
+			}
+
+			if tc.err == nil {
+				defer resetWorker()
+			}
+
+			if got != nil {
+				assert.NotNilf(t, tc.res, "Expected UpdareWorker response to be nil, but was %v", got)
```
Comment (irenarindos) [inline internal/daemon/controller/handlers/workers/worker_service_test.go:644]:
```suggestion
				assert.NotNilf(t, tc.res, "Expected UpdateWorker response to be nil, but was %v", got)
```

File: internal/proto/controller/api/resources/workers/v1/worker.proto [hunk 2]
```diff
@@ -62,7 +62,13 @@ message Worker {
...
 
   // The address that this worker is reachable at.
-  google.protobuf.StringValue address = 90; // @gotags: `class:"public"`
+  google.protobuf.StringValue address = 90 [
+    (custom_options.v1.generate_sdk_option) = true,
+    (custom_options.v1.mask_mapping) = {
+      this: "address"
+      that: "address"
+    }
+  ]; // @gotags: `class:"sensitive"`
```
Comment (jimlambrt) [inline internal/proto/controller/api/resources/workers/v1/worker.proto:71]: a worker IP is sensitive?

File: internal/daemon/controller/handlers/workers/worker_service.go [hunk 3]
```diff
@@ -393,3 +473,22 @@ func validateListRequest(req *pbs.ListWorkersRequest) error {
...
 }
+
+func validateUpdateRequest(req *pbs.UpdateWorkerRequest) error {
+	return handlers.ValidateUpdateRequest(req, req.GetItem(), func() map[string]string {
+		badFields := map[string]string{}
+		if req.GetItem().GetWorkerConfig() != nil {
+			badFields[globals.ConfigurationField] = "This is a read only field."
+		}
+		if req.GetItem().GetCanonicalAddress() != "" {
+			badFields[globals.CanonicalAddressField] = "This is a read only field."
```
Comment (jimlambrt) [inline internal/daemon/controller/handlers/workers/worker_service.go:484]: perhaps add unit tests for badFields for: CanonicalAddress, CanonicalTags and Tags?

**Review by jimlambrt — CHANGES_REQUESTED** [review]
a few minor questions.

**Review by jimlambrt — APPROVED** [review]

**Pull request merged**